ndt7.go

```diff
@@ -170,10 +185,32 @@ func (c *Client) start(ctx context.Context, f testFn, p string) (<-chan spec.Mea
 		return nil, err
 	}
 	ch := make(chan spec.Measurement)
-	go f(ctx, conn, ch)
+	go c.collectData(ctx, f, conn, ch)
 	return ch, nil
 }
 
+func (c *Client) collectData(ctx context.Context, f testFn, conn websocketx.Conn, outch chan<- spec.Measurement) {
+	inch := make(chan spec.Measurement)
+	defer close(outch)
+	go f(ctx, conn, inch)
+
+	for m := range inch {
+		switch m.Origin {
+		case spec.OriginClient:
+			c.results[m.Test].Client = m
+		case spec.OriginServer:
+			// The server only sends ConnectionInfo once at the beginning of
+			// the test, thus if we want to know the client IP and test UUID
+			// we need to store it separately.
```

Comment:
Edited https://github.com/m-lab/ndt-server/issues/194#issuecomment-539100964 such that we remember to change this code here.